netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -85,25 +87,30 @@
   private final ServerTransportListener transportListener;
   private final int maxMessageSize;
   private Throwable connectionError;
-  private ChannelHandlerContext ctx;
   private boolean teWarningLogged;
-  private int connectionWindow;
   private WriteQueue serverWriteQueue;
 
   NettyServerHandler(ServerTransportListener transportListener,
-      Http2Connection connection,
-      Http2FrameReader frameReader,
-      Http2FrameWriter frameWriter,
-      int maxStreams,
-      int flowControlWindow,
-      int maxMessageSize) {
-    super(connection, frameReader, frameWriter, new LazyFrameListener(),
-          createInitialSettings(flowControlWindow, maxStreams));
-    this.connectionWindow = flowControlWindow;
+                     Http2Connection connection,
+                     Http2FrameReader frameReader,
+                     Http2FrameWriter frameWriter,
+                     int maxStreams,
+                     int flowControlWindow,
+                     int maxMessageSize) {
+    this(transportListener, new DefaultHttp2ConnectionEncoder(connection, frameWriter), frameReader,
+        createInitialSettings(flowControlWindow, maxStreams), maxMessageSize);
+  }
+
+  private NettyServerHandler(ServerTransportListener transportListener,
```

Comment:
Need to create the encoder and the decoder ... but the decoder depends on the encoder.